test/core/libp2p/muxer/yamux/yamux_integration_test.cpp

```diff
@@ -0,0 +1,400 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/muxer/yamux/yamux.hpp"
+
+#include <gtest/gtest.h>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/muxer/yamux/yamux_frame.hpp"
+#include "libp2p/muxer/yamux/yamux_stream.hpp"
+#include "libp2p/transport/all.hpp"
+#include "testutil/outcome.hpp"
+
+using namespace libp2p::muxer;
+using namespace libp2p::transport;
+using namespace libp2p::stream;
+using namespace kagome::common;
+using namespace libp2p::multi;
+
+using std::chrono_literals::operator""ms;
+
+class YamuxIntegrationTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    init();
+  }
+
+  /**
+   * Initialize:
+   *    - transport
+   *    - listener, which wraps new connections into Yamux
+   *    - multiaddress, which is listened by the created transport
+   *    - Yamux itself
+   */
+  void init() {
+    // create transport
+    transport_ = std::make_unique<TransportImpl>(context_);
+    ASSERT_TRUE(transport_) << "cannot create transport";
+
+    // create a listener, which is going to wrap new connections to Yamux
+    transport_listener_ = transport_->createListener(
+        [this](std::shared_ptr<Connection> c) mutable {
+          ASSERT_FALSE(c->isClosed());
+          ASSERT_TRUE(c) << "createListener: connection is nullptr";
+
+          // here, our Yamux instance is going to be a server, as a new
+          // connection is accepted
+          yamux_ = std::make_shared<Yamux>(
+              std::move(c),
+              [this](std::unique_ptr<Stream> new_stream) mutable {
+                accepted_streams_.push_back(std::move(new_stream));
+              },
+              YamuxConfig{true});
+
+          ASSERT_TRUE(yamux_) << "cannot create Yamux from a new connection";
+          yamux_->start();
+        });
+
+    // create multiaddress, from which we are going to connect
+    EXPECT_OUTCOME_TRUE(ma, Multiaddress::create("/ip4/127.0.0.1/tcp/40009"))
+    EXPECT_TRUE(transport_listener_->listen(ma)) << "is port 40009 busy?";
+    multiaddress_ = std::make_shared<Multiaddress>(std::move(ma));
+
+    // dial to our "server", getting a connection
+    EXPECT_OUTCOME_TRUE(conn, transport_->dial(*multiaddress_))
+    connection_ = std::move(conn);
+
+    // wait until yamux gets created from that connection
+    context_.run_for(10ms);
```

Comment:
it does not wait, it starts event loop for 10 ms and if there is no work, it exits (or exits in 10 ms)

Reply:
Misunderstanding of the comment. Reformulated it